src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectSampleService.java

```diff
@@ -40,6 +44,29 @@ public UIProjectSampleService(ProjectService projectService, SampleService sampl
 		this.messageSource = messageSource;
 	}
 
+	/**
+	 * Validate a list of sample names
+	 *
+	 * @param projectId project identifier
+	 * @param request   {@link ValidateSampleNamesRequest} details about the sample names to validate
+	 * @return a list of validated sample names
+	 */
+	public ValidateSampleNamesResponse validateSampleNames(Long projectId, ValidateSampleNamesRequest request) {
+		List<ValidateSampleNameModel> samples = request.getSamples();
+		List<Long> associatedProjectIds = request.getAssociatedProjectIds();
+		List<Long> projectIds = new ArrayList<>();
+		projectIds.add(projectId);
+		if (associatedProjectIds != null) {
+			projectIds.addAll(associatedProjectIds);
+		}
+		for (ValidateSampleNameModel sample : samples) {
+			String sampleName = sample.getName();
+			List<Long> foundSampleIds = sampleService.getSamplesBySampleNameForProjects(projectIds, sampleName);
```

Comment:
Can we change this to a single service call. That way we can only execute 1 query vs n queries. If you need an example of how to do this you can look at PR #1360.

Reply:
Like this? 3510e4466f06ce6b5f1cb549dfe18318e09bf01b

Reply:
Yup that looks good.